pkg/git/lsremote.go

```diff
@@ -229,6 +228,10 @@ func (g *git) setCredential(cred *corev1.Secret) error {
 
 func (g *git) gitCmd(output io.Writer, subCmd string, args ...string) error {
 	kv := fmt.Sprintf("credential.helper=%s", `/bin/sh -c 'echo "password=$GIT_PASSWORD"'`)
+	//nolint:gosec // this exec deals with user input:
+	// $GIT_PASSWORD, g.URL, branch, commit
+	// The args are validated before and the -- is used to help git
```

Comment:
What I understand from this is that `gitCmd` assumes that `args` have been validated by the caller.
Is that understanding correct? If so, should that assumption be mentioned in function-level doc then?

Reply:
It's not just args, there also user controlled fields on the struct.
The function is safe without the validation. The validation is just there to be more defensive.